### PR TITLE
fix(docs): Update markdown processor to use GFM

### LINKED ISSUE #5559: GHP bad rendering on CONTRIBUTE & HOWTO (fa_IR locale)
## Problem

It seems that Github Pages engine doesn't deal fine rendering links for this RTL resources:

- https://ebookfoundation.github.io/free-programming-books/HOWTO-fa_IR.html
  image
- https://ebookfoundation.github.io/free-programming-books/CONTRIBUTE-fa_IR.html
  image

It **works well in other markdown files** like books, courses... for the same locale.

## Solution

Maybe due to some issue with the `div` tag:
https://github.com/EbookFoundation/free-programming-books/blob/e629db6b05a72fd64a6b33feeeebbe830641cf10/HOWTO-fa_IR.md#L1-L5

Need some experiments to deal with this tag, and maybe apply on listings too.

### PR DESCRIPTION
**Pull request by charlottetan**
## What does this PR do?
Improve repo: Update the markdown processor to use GFM instead of the default (kramdown). GFM is the processor Github uses. Resolves #5559.

#### Docs for more information:
https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll

#### Examples here:
https://charlottetan.github.io/free-programming-books/CONTRIBUTING-fa_IR.html

https://charlottetan.github.io/free-programming-books/HOWTO-fa_IR.html

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [ ] Search for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by davorpa** [comment]
Update pr title with scope `fix(docs): ....`

**Comment by davorpa** [comment]
More docs about GMF & Kramdown: https://jekyllrb.com/docs/configuration/markdown/#commonmark

:ok: I have not seen any other listing break.

File: _config.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # [Name of visual theme]
 #theme: jekyll-theme-minimal
 remote_theme: pages-themes/minimal@v0.2.0
+markdown: GFM
```
Comment (davorpa) [inline _config.yml:4]: Add documentation comments (a new line between sections)

```markdown
# [Conversion]
markdown: GFM
```

It's a bit wird why use directly `GFM` is a bit different of `kramdown` with `input: GFM` (default). Maybe, to investigate about it and report to Jekyll project @jekyll.

From default config docs: https://jekyllrb.com/docs/configuration/default/
```yml
# Markdown Processors
kramdown:
  auto_ids          : true
  entity_output     : as_char
  toc_levels        : [1, 2, 3, 4, 5, 6]
  smart_quotes      : lsquo,rsquo,ldquo,rdquo
  input             : GFM
  hard_wrap         : false
  footnote_nr       : 1
  show_warnings     : false
```

Comment (davorpa) [inline _config.yml:4]: :warning: Add documentation comments (a new line between sections)

```markdown
# [Conversion]
markdown: GFM
```

---

It's a bit weird why use directly `GFM` is a bit different of `kramdown` with `input: GFM` (default). Maybe, to investigate about it and report to Jekyll project @jekyll.

From default config docs: https://jekyllrb.com/docs/configuration/default/
```yml
# Markdown Processors
kramdown:
  auto_ids          : true
  entity_output     : as_char
  toc_levels        : [1, 2, 3, 4, 5, 6]
  smart_quotes      : lsquo,rsquo,ldquo,rdquo
  input             : GFM
  hard_wrap         : false
  footnote_nr       : 1
  show_warnings     : false
```

**Review by davorpa — CHANGES_REQUESTED** [review]

File: _config.yml (hunk 1 above)
Comment (charlottetan) [reply _config.yml:6]: Hmm looks like it's just two different implementations: https://jekyllrb.com/docs/configuration/markdown/ (doc also available on [github](https://github.com/jekyll/jekyll/blob/master/docs/_docs/configuration/markdown.md), which I find easier to read). The two implementations being [kramdown/parser-gfm](https://github.com/kramdown/parser-gfm) vs [github/cmark-gfm](https://github.com/github/cmark-gfm).

Looking at [kramdown/parser-gfm's issue list](https://github.com/kramdown/parser-gfm/issues), it looks like there are some existing differences. I've filed a new issue here: 
[kramdown/parser-gfm#28: Parser doesn't process markdown nested inside of HTML tags](https://github.com/kramdown/parser-gfm/issues/28)

I've added documentation comments to the open PR.

**Review by davorpa — APPROVED** [review]
### Ready to merge :fireworks: :tada:

Thanks for your valuable researches

**Comment by eshellman** [comment]
Awesome teamwork

**Pull request merged**